bloom/generators/common.py

```diff
@@ -133,6 +144,22 @@ def package_conditional_context(ros_distro):
             'ROS_DISTRO': ros_distro,
             }
 
+def evaluate_package_conditions(package, ros_distro):
+    """
+    Evaluates a package's conditional fields.
+
+    :param package: The package to be evaluated.
+    :param ros_distro: The codename of the rosdistro use for context.
+    :returns: None. The given package will be modified.
+    """
+    # Conditional fields were introduced in package format 3.
+    # Earlier formats should have their conditions evaluated with no context so
+    # the evaluated_condition is set to True in all cases.
+    if package.package_format < 3:
+        package.evaluate_conditions({})
```

Comment:
That doesn't seem to make sense. Why would this be needed if the format doesn't even have conditions?

Reply:
I had initially done this so that evaluated_condition was true on all cases so that the generators would not need to duplicate the package version check.

I've since looked at [a condition check in catkin_pkg](https://github.com/ros-infrastructure/catkin_pkg/blob/7b889b2153485943ad3eb19e824059c5d04edb2d/src/catkin_pkg/package.py#L155) and switched Bloom to using the same `evaluated_condition is not False` check to allow for an evaluated_condition of None.

I think there's some risk to doing this in general because until evaluate_conditions() is invoked all evaluated_condition values are `None`. So attention must be paid to ensure that evaluate_conditions is always invoked when needed.